fbw-common/src/wasm/systems/systems/src/communications/receivers.rs

```diff
@@ -0,0 +1,307 @@
+use uom::num::pow;
+
+pub struct CommTransceiver {
+    is_power_supply_powered: bool,
+    powered_by: ElectricalBusType,
+}
+impl CommTransceiver {
+    pub fn new(powered_by: ElectricalBusType) -> Self {
+        Self {
+            is_power_supply_powered: false,
+            powered_by,
+        }
+    }
+
+    pub fn is_powered(&self) -> bool {
+        self.is_power_supply_powered
+    }
+}
+
+impl SimulationElement for CommTransceiver {
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_power_supply_powered = buses.is_powered(self.powered_by);
+    }
+}
+
+pub struct NavReceiver {
+    beep_id: VariableIdentifier,
+
+    is_power_supply_powered: bool,
+    powered_by: ElectricalBusType,
+
+    morse: Morse,
+
+    ok_to_beep: bool,
+}
+impl NavReceiver {
+    pub fn new(
+        context: &mut InitContext,
+        name: &str,
+        id: usize,
+        powered_by: ElectricalBusType,
+    ) -> Self {
+        Self {
+            beep_id: context.get_identifier(format!("ACP_BEEP_IDENT_{}{}", name, id)),
+            is_power_supply_powered: false,
+            powered_by,
+            morse: Morse::new(context, name, id),
+            // Always true to VORs and ADFs.
+            // Used for ILS
+            // Called in update()
+            ok_to_beep: false,
+        }
+    }
+
+    pub fn is_powered(&self) -> bool {
+        self.is_power_supply_powered
+    }
+
+    pub fn update(&mut self, context: &UpdateContext, ok_to_beep: bool) {
+        // We keep updating the morse even though the receiver is not powered
+        // because in real life, the signal is external (obviously)
+        self.ok_to_beep = ok_to_beep;
+        self.morse.update(context);
+    }
+}
+
+impl SimulationElement for NavReceiver {
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(
+            &self.beep_id,
+            if self.is_power_supply_powered && self.ok_to_beep {
+                self.morse.get_state()
+            } else {
+                false
+            },
+        );
+    }
+
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_power_supply_powered = buses.is_powered(self.powered_by);
+    }
+}
+
+// #[derive(Clone)]
+// pub struct MARKERS {
+//     is_power_supply_powered: bool,
+//     powered_by: ElectricalBusType,
+// }
+// impl MARKERS {
+//     pub fn new_markers() -> Self {
+//         Self {
+//             is_power_supply_powered: false,
+//             powered_by: ElectricalBusType::DirectCurrent(1),
+//         }
+//     }
+
+//     pub fn is_powered(&self) -> bool {
+//         self.is_power_supply_powered
+//     }
+// }
+
+// impl SimulationElement for MARKERS {
+//     fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+//         visitor.visit(self);
+//     }
+
+//     fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+//         self.is_power_supply_powered = buses.is_powered(self.powered_by);
+//     }
+// }
```

Comment:
I assume now that the SDK has been done this can also be uncommented?

Reply:
I'd like to keep it as-is so I just need to remove the comment.

I'll add a comment on top.

Reply:
I've removed it. I'll put the markers as nav receivers and modify morse generation code as required.